# Beneficial Apps of App Store and Google Play Markets

For this project, we'll pretend we're working as data analyst for a company that builds Android and iOS mobile apps.

We only build apps that are free to download and install, and our main source of revenue consists of in-app ads. This means our revenue for any given app is mostly influenced by the number of users who use our app. Our goal for this project is to analyze data to help our developers understand what kinds of apps are likely to attract more users.  

To do this, we'll need to collect and analyze data about mobile apps available on Google Play and the App Store.

## Opening and explore data

Collecting data for over four million apps requires a significant amount of time and money, so we'll try to analyze a sample of the data instead. To avoid spending resources on collecting new data ourselves, we should first try to see whether we can find any relevant existing data at no cost. Luckily, these are two data sets that seem suitable for our goals:  
* [A data set](https://www.kaggle.com/lava18/google-play-store-apps/home) containing data about approximately ten thousand Android apps from Google Play — the data was collected in August 2018
* [A data set](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/home) containing data about approximately seven thousand iOS apps from the App Store — the data was collected in July 2017

We'll start by opening and exploring these two data sets.

In [127]:
opened_ios_file = open('AppleStore.csv')
from csv import reader
readed_ios = reader(opened_ios_file)
data_set_ios = list(readed_ios)
ios_header = data_set_ios[0]
ios = data_set_ios[1:]


opened_android_file = open('googleplaystore.csv')
from csv import reader
readed_android = reader(opened_android_file)
data_set_android = list(readed_android)
android_header = data_set_android[0]
android = data_set_android[1:]

def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

In [128]:
print(ios_header)
print('\n')
explore_data(ios, 0, 3, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 7197
Number of columns: 16


We see, that ios applications have 7197 apps.  
The names as: 
* "track_name": App Name
* "size_bytes": Size (in Bytes)
* "currency": Currency Type
* "price": Price amount
* "user_rating_ver": Average User Rating value (for current version)
* "cont_rating": Content Rating
* "prime_genre": Primary Genre   

are may be interesting for us.
All column name description you may see [here](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/home)

In [129]:
print(android_header)
print('\n')
explore_data(android, 0, 3, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13


We see, that android applications have 10841 apps.  
The names as: 
* 'App'
* 'Rating'
* 'Size'
* 'Price'
* 'Content Rating'
* 'Genres'  

are may be interesting for us.

# Deleting wrong data (Data cleaning)

We opened the two data sets and performed a brief exploration of the data. Before beginning our analysis, we need to make sure the data we analyze is accurate, otherwise the results of our analysis will be wrong. This means that we'll need to:

* Detect inaccurate data and correct (or remove) it
* Detect duplicate data and remove the duplicates

Our company we only build apps that are free to download and install, and that are directed toward an English-speaking audience. This means that we'll need to:

* Remove non-English apps like 爱奇艺PPS -《欢乐颂2》电视剧热播
* Remove non-free apps

The Google Play data set has a dedicated [discussion section](https://www.kaggle.com/lava18/google-play-store-apps/discussion), and we can see that [one of the discussions](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015) describes an error for a certain row 10472 row has missing 'Rating' and a column shift happened for next columns. 


In [130]:
print(android_header)
print('\n')
print(android[10471]) # correct row
print('\n')
print(android[10472]) #incorrect row

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Xposed Wi-Fi-Pwd', 'PERSONALIZATION', '3.5', '1042', '404k', '100,000+', 'Free', '0', 'Everyone', 'Personalization', 'August 5, 2014', '3.0.0', '4.0.3 and up']


['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


We're deleting this wrong row

In [131]:
print(len(android))
del android[10472]
print(len(android))

10841
10840


# Deleting duplicate rows

We started the data cleaning process and deleted a row with incorrect data from the Google Play data set. But according to the discussions section, we notice that some apps have duplicate entries. For instance, Instagram has four entries.
We analyse duplicate rows.

In [132]:
for apps in android:
    name = apps[0]
    if name == 'Quick PDF Scanner + OCR FREE':
        print(apps)

['Quick PDF Scanner + OCR FREE', 'BUSINESS', '4.2', '80805', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'February 26, 2018', 'Varies with device', '4.0.3 and up']
['Quick PDF Scanner + OCR FREE', 'BUSINESS', '4.2', '80805', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'February 26, 2018', 'Varies with device', '4.0.3 and up']
['Quick PDF Scanner + OCR FREE', 'BUSINESS', '4.2', '80804', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'February 26, 2018', 'Varies with device', '4.0.3 and up']


As we see, that 'Quick PDF Scanner + OCR FREE' is three times in data set

In [133]:
duplicate_rows_android = []
unique_rows_android = []

for row in android:
    name = row[0]
    if name in unique_rows_android:
        duplicate_rows_android.append(name)
    else:
        unique_rows_android.append(name)
print(len(duplicate_rows_android))
print(len(unique_rows_android))

1181
9659


In [134]:
duplicate_rows_ios = []
unique_rows_ios = []

for row in ios:
    name = row[1]
    if name in unique_rows_ios:
        duplicate_rows_ios.append(name)
    else:
        unique_rows_ios.append(name)
print(len(duplicate_rows_ios))
print(len(unique_rows_ios))
print(duplicate_rows_ios)

2
7195
['Mannequin Challenge', 'VR Roller Coaster']


We find two dublicate rows in App store dataset. Let's analyse one of it.

In [135]:
print(ios_header)
print('\n')
for apps in ios:
    name = apps[1]
    if name == 'Mannequin Challenge':
        print(apps)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['1173990889', 'Mannequin Challenge', '109705216', 'USD', '0.0', '668', '87', '3.0', '3.0', '1.4', '9+', 'Games', '37', '4', '1', '1']
['1178454060', 'Mannequin Challenge', '59572224', 'USD', '0.0', '105', '58', '4.0', '4.5', '1.0.1', '4+', 'Games', '38', '5', '1', '1']


Above, we:

* Created two lists: one for storing the name of duplicate apps, and one for storing the name of unique apps
* Looped through the `android` data set (the Google Play data set) and `ios` data set (the Apple store data set), and for each iteration, we:
 * Appended the app name to the `duplicate_apps` list if the app name was already in the `unique_apps` list
 * Appended the app name to the `unique_apps` list, if the app name wasn't already in the `unique_apps` list

We don't want to count certain apps more than once when we analyze data, so we need to remove the duplicate entries and keep only one entry per app. One thing we could do is remove the duplicate rows randomly, but we could probably find a better way.

If you examine the rows of android data set we printed above for the Quick PDF Scanner + OCR FREE app, the main difference happens on the fourth position of each row, which corresponds to the number of reviews.

If you examine the rows of ios data set we printed above for the Mannequin Challenge app, the main difference happens on the fifth position of each row, which corresponds to the rating_count_tot. 

The different numbers show that the data was collected at different times.

We could use this information to build a criterion for removing the duplicates. 

The higher the number of reviews in android data set and User Rating counts (for all version) in ios data set, the more recent the data should be. Rather than removing duplicates randomly, we'll only keep the row with the highest number of reviews / User Rating counts and remove the other entries for any given app.

To remove the duplicates in android data set, we will:

* Create a dictionary, where each dictionary key is a unique app name and the corresponding dictionary value is the highest number of reviews of that app
* Use the information stored in the dictionary to create a new data set, which will have only one entry per app (and for each app, we'll only select the entry with the highest number of reviews)

In [136]:
reviews_max = {}
for row in android:
    name = row[0]
    n_reviews = float(row[3])
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    elif name not in reviews_max:
        reviews_max[name] = n_reviews
print(len(reviews_max))

9659


As we expected, we got the `reviews_max` dictionary with 9659 entries.  
We use the dictionary created above to remove the duplicate rows.

Start by creating two empty lists: `android_clean` (which will store our new cleaned data set) and `already_added_android` (which will just store app names).

In [137]:
android_clean = []
already_added_android = []
for row in android:
    name = row[0]
    n_reviews = float(row[3])
    if reviews_max[name] == n_reviews and name not in already_added_android:
        android_clean.append(row)
        already_added_android.append(name)
print(len(android_clean))

9659


As we explore the `android_clean` data set has 9,659 rows.

To remove the duplicates in ios data set, we will:

* Create a dictionary, where each dictionary key is a unique app name and the corresponding dictionary value is the highest User Rating counts of that app
* Use the information stored in the dictionary to create a new data set, which will have only one entry per app (and for each app, we'll only select the entry with the highest number of reviews)

In [138]:
ratings_max = {}
for row in ios:
    name = row[1]
    n_rating = float(row[5])
    if name in ratings_max and ratings_max[name] < n_rating:
        ratings_max[name] = n_rating
    elif name not in ratings_max:
        ratings_max[name] = n_rating
print(len(ratings_max))

7195


As we expected, we got the `ratings_max` dictionary with 7195 entries.
We use the dictionary created above to remove the duplicate rows.

Start by creating two empty lists: ios_clean (which will store our new cleaned data set) and already_added_ios (which will just store app names).

In [139]:
ios_clean = []
already_added_ios = []
for row in ios:
    name = row[1]
    n_rating = float(row[5])
    if ratings_max[name] == n_rating and name not in already_added_ios:
        ios_clean.append(row)
        already_added_ios.append(name)
print(len(ios_clean))

7195


As we explore the `ios_clean` data set has 7,195 rows.

# Deleting non-english apps

Remember that the language we use for our apps is English, and we'd like to analyze only the apps that are directed toward an English-speaking audience. However, if we explore the data long enough, we'll find that both data sets have apps whose name suggests that they are not direct toward an English-speaking audience.

In [140]:
print(ios_clean[813][1])
print(ios_clean[6731][1])
print('\n')
print(android_clean[4412][0])

爱奇艺PPS -《欢乐颂2》电视剧热播
激ムズ！和のひとふで書き！ 〜頭をつかう脳トレパズルゲーム〜


中国語 AQリスニング


We're not interested in keeping these kind of apps, so we'll remove them. One way to go about this is to remove each app whose name contains a symbol that is not commonly used in English text — English text usually includes letters from the English alphabet, numbers composed of digits from 0 to 9, punctuation marks (., !, ?, ;, etc.), and other symbols (+, *, /, etc.).

Behind the scenes, each character we use in a string has a corresponding number associated with it.  
We can get the corresponding number of each character using the `ord()` [built-in function](https://docs.python.org/3/library/functions.html#ord).  

The numbers corresponding to the characters we commonly use in an English text are all in the range 0 to 127, according to the [ASCII](https://en.wikipedia.org/wiki/ASCII) (American Standard Code for Information Interchange) system. Based on this number range, we can build a function that detects whether a character belongs to the set of common English characters or not. 
But emojis and some characters like ™ fall outside the ASCII range and have corresponding numbers that are over 127.  
To minimize the impact of data loss, we'll only remove an app if its name has more than three characters with corresponding numbers falling outside the ASCII range.  
Firstly we will define a function which will find non-english apps:

In [141]:
def english_char(string):
    non_english = 0
    for character in string:
        if ord(character) > 127:
            non_english += 1
    if non_english > 3:
        return False
    return True

#Test_A
print(english_char('Instagram'))

#Test_B
print(english_char('爱奇艺PPS -《欢乐颂2》电视剧热播'))

#Test_C
print(english_char('Docs To Go™ Free Office Suite'))

#Test_D
print(english_char('Instachat 😜'))

True
False
True
True


To filter out non-English apps from both data sets (ios and android). We loop through each data set, and if an app name is identified as English, then append the whole row to a separate list.

In [142]:
eng_ios =[]
for row in ios_clean:
    name = row[1]
    if english_char(name) == True:
        eng_ios.append(row)
print(len(eng_ios))

6181


As we see ios dataset has 6181 english apps.  
Let's check the android dataset.

In [143]:
eng_android =[]
for row in android_clean:
    name = row[0]
    if english_char(name) == True:
        eng_android.append(row)
print(len(eng_android))

9614


As we see 45 android apps is not english and we deleted it from our analysing data.  

# Separating free apps

As we mentioned in the introduction, we only build apps that are free to download and install, and our main source of revenue consists of in-app ads. Our data sets contain both free and non-free apps, and we'll need to isolate only the free apps for our analysis.  
We will loop through each data set to isolate the free apps in separate lists.

In [144]:
free_eng_ios =[]
for row in eng_ios:
    if row[4] == '0.0':
        free_eng_ios.append(row)

print(len(free_eng_ios))

free_eng_android =[]
for row in eng_android:
    if row[7] == '0':
        free_eng_android.append(row)

print(len(free_eng_android))

3220
8864


As we see, only 3220 ios apps and 8864 android apps are free.

# Most common apps by genre.

As we mentioned in the introduction, our aim is to determine the kinds of apps that are likely to attract more users because our revenue is highly influenced by the number of people using our apps.

To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:

1. Build a minimal Android version of the app, and add it to Google Play.
2. If the app has a good response from users, we then develop it further.
3. If the app is profitable after six months, we also build an iOS version of the app and add it to the App Store.  

Because our end goal is to add the app on both the App Store and Google Play, we need to find app profiles that are successful on both markets. For instance, a profile that might work well for both markets might be a productivity app that makes use of gamification.

Let's begin the analysis by getting a sense of what are the most common genres for each market. For this, we'll build a frequency table for the prime_genre column of the App Store data set, and the Genres and Category columns of the Google Play data set.

We'll build two functions we can use to analyze the frequency tables:

* One function to generate frequency tables that show percentages
* Another function we can use to display the percentages in a descending order

In [145]:
def freq_table(dataset, index):
    frequency_table = {}
    total = 0
    for row in dataset:
        total += 1
        genre = row[index]
        if genre in frequency_table:
            frequency_table[genre] += 1
        else:
            frequency_table[genre] = 1
    
    for app in frequency_table:
        frequency_table[app] = (frequency_table[app] / total) * 100
    
    return frequency_table
     
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

We'll analyze the frequency table for the prime_genre column of the App Store data set

In [146]:
display_table(free_eng_ios, 11)

Games : 58.13664596273293
Entertainment : 7.888198757763975
Photo & Video : 4.968944099378882
Education : 3.6645962732919255
Social Networking : 3.291925465838509
Shopping : 2.608695652173913
Utilities : 2.515527950310559
Sports : 2.142857142857143
Music : 2.049689440993789
Health & Fitness : 2.018633540372671
Productivity : 1.7391304347826086
Lifestyle : 1.5838509316770186
News : 1.3354037267080745
Travel : 1.2422360248447204
Finance : 1.1180124223602486
Weather : 0.8695652173913043
Food & Drink : 0.8074534161490683
Reference : 0.5590062111801243
Business : 0.5279503105590062
Book : 0.43478260869565216
Navigation : 0.18633540372670807
Medical : 0.18633540372670807
Catalogs : 0.12422360248447205


Table shows us that the most common genre of the App Store data set is Games, about 58%. Entertainment is second place, has almost 8%, followed by photo and video apps, which are close to 5%. Only 3.66% of the apps are designed for education, followed by social networking apps which amount for 3.29% of the apps in our data set.

The general impression is that App Store (at least the part containing free English apps) is dominated by apps that are designed for fun (games, entertainment, photo and video, social networking, sports, music, etc.), while apps with practical purposes (education, shopping, utilities, productivity, lifestyle, etc.) are more rare. However, the fact that fun apps are the most numerous doesn't also imply that they also have the greatest number of users — the demand might not be the same as the offer.

Let's continue by examining the Genres and Category columns of the Google Play data set (two columns which seem to be related).

In [147]:
display_table(free_eng_android, 1) # category

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

Table shows us that the most common category of the Google Play data set is Family, about 18.9%. Games is second place, has 9.72%, followed by tools, which are close to 8.5%. Only 4.59% of the apps are designed for business, followed by lifestyle apps which amount for 3.9% of the apps in our data set.  
We see significantly different on Google Play: there are not that many apps designed for fun, and it seems that a good number of apps are designed for practical purposes (family, tools, business, lifestyle, productivity, etc.)  
Practical apps seem to have a better representation on Google Play compared to App Store. This picture is also confirmed by the frequency table we see for the Genres column:

In [148]:
display_table(free_eng_android, 9) # genre

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

# Most Popular Apps by Genre on the App Store

The frequency tables we analyzed showed us that the App Store is dominated by apps designed for fun, while Google Play shows a more balanced landscape of both practical and for-fun apps. Now, we'd like to get an idea about the kind of apps with the most users.  
One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. For the Google Play data set, we can find this information in the Installs column, but this information is missing for the App Store data set. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the rating_count_tot app. 

Let's start with calculating the average number of user ratings per app genre on the App Store. To do that, we'll need to:

* Isolate the apps of each genre
* Sum up the user ratings for the apps of that genre
* Divide the sum by the number of apps belonging to that genre (not by the total number of apps)

In [149]:
unique_app_genres = freq_table(free_eng_ios, -5)
for genre in unique_app_genres:
    total = 0
    len_genre = 0
    for row in free_eng_ios:
        genre_app = row[-5]
        if genre_app == genre:
            total += float(row[5])
            len_genre += 1
    aver_of_users_rat = total / len_genre
    print(genre, ':', aver_of_users_rat)

Utilities : 18684.456790123455
Photo & Video : 28441.54375
Book : 39758.5
Business : 7491.117647058823
Food & Drink : 33333.92307692308
Productivity : 21028.410714285714
Catalogs : 4004.0
Medical : 612.0
Health & Fitness : 23298.015384615384
Games : 22812.92467948718
Shopping : 26919.690476190477
Education : 7003.983050847458
Music : 57326.530303030304
Reference : 74942.11111111111
News : 21248.023255813954
Entertainment : 14029.830708661417
Social Networking : 71548.34905660378
Finance : 31467.944444444445
Weather : 52279.892857142855
Sports : 23008.898550724636
Travel : 28243.8
Lifestyle : 16485.764705882353
Navigation : 86090.33333333333


As we see above according to User Rating counts of free apps in Apple Store, most popular genres are Navigation, Social Networking, Reference, Weather, Music. Almost all of them are built for practical purposes.
That means the most popular apps are differ from most common in App store.

But let's analyse which apps have the most popular genres.
For instance, Navigation, Social Networking, Reference.

In [150]:
for app in free_eng_ios:
    if app[-5] == 'Navigation':
        print(app[1], ':', app[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


It means the only two apps such as Waze, Google Maps have huge amount of all app from Navigation genre.

In [151]:
for app in free_eng_ios:
    if app[-5] == 'Social Networking':
        print(app[1], ':', app[5])

Facebook : 2974676
Pinterest : 1061624
Skype for iPhone : 373519
Messenger : 351466
Tumblr : 334293
WhatsApp Messenger : 287589
Kik : 260965
ooVoo – Free Video Call, Text and Voice : 177501
TextNow - Unlimited Text + Calls : 164963
Viber Messenger – Text & Call : 164249
Followers - Social Analytics For Instagram : 112778
MeetMe - Chat and Meet New People : 97072
We Heart It - Fashion, wallpapers, quotes, tattoos : 90414
InsTrack for Instagram - Analytics Plus More : 85535
Tango - Free Video Call, Voice and Chat : 75412
LinkedIn : 71856
Match™ - #1 Dating App. : 60659
Skype for iPad : 60163
POF - Best Dating App for Conversations : 52642
Timehop : 49510
Find My Family, Friends & iPhone - Life360 Locator : 43877
Whisper - Share, Express, Meet : 39819
Hangouts : 36404
LINE PLAY - Your Avatar World : 34677
WeChat : 34584
Badoo - Meet New People, Chat, Socialize. : 34428
Followers + for Instagram - Follower Analytics : 28633
GroupMe : 28260
Marco Polo Video Walkie Talkie : 27662
Miitomo : 2

Facebook and Pinterest take the most of rating numbers.

In [152]:
for app in free_eng_ios:
    if app[-5] == 'Reference':
        print(app[1], ':', app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


Bible and Dictionary.com take the most of rating numbers.
It means that Apple store app are still most popular in for fun segment. And only few apps from 'for practical purposes' have users popularity.

# Most Popular Apps by Genre on the Google Play

We have data about the number of installs for the Google Play market, so we should be able to get a clearer picture about genre popularity. However, the install numbers don't seem precise enough — we can see that most values are open-ended (100+, 1,000+, 5,000+, etc.):

In [153]:
display_table(free_eng_android, 5)

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


One problem with this data is that it's not precise. For instance, we don't know whether an app with 100,000+ installs has 100,000 installs, 200,000, or 350,000. However, we don't need very precise data for our purposes — we only want to find out which app genres attract the most users, and we don't need perfect precision with respect to the number of users.

We're going to leave the numbers as they are, which means that we'll consider that an app with 100,000+ installs has 100,000 installs, and an app with 1,000,000+ installs has 1,000,000 installs, and so on. To perform computations, however, we'll need to convert each install number from string to float. This means we need to remove the commas and the plus characters, otherwise the conversion will fail and raise an error.

In [154]:
uniq_app_category = freq_table(free_eng_android, 1)
for category in uniq_app_category:
    total = 0
    len_category = 0
    for app in free_eng_android:
        category_app = app[1]
        if category == category_app:
            n_installs = app[5]
            n_installs = n_installs.replace('+', '')
            n_installs = n_installs.replace(',', '')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total / len_category
    print(category, ':', avg_n_installs)

COMMUNICATION : 38456119.167247385
BEAUTY : 513151.88679245283
LIFESTYLE : 1437816.2687861272
FINANCE : 1387692.475609756
PHOTOGRAPHY : 17840110.40229885
SOCIAL : 23253652.127118643
EVENTS : 253542.22222222222
MAPS_AND_NAVIGATION : 4056941.7741935486
MEDICAL : 120550.61980830671
SHOPPING : 7036877.311557789
DATING : 854028.8303030303
BUSINESS : 1712290.1474201474
VIDEO_PLAYERS : 24727872.452830188
ENTERTAINMENT : 11640705.88235294
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
AUTO_AND_VEHICLES : 647317.8170731707
FOOD_AND_DRINK : 1924897.7363636363
LIBRARIES_AND_DEMO : 638503.734939759
FAMILY : 3695641.8198090694
PERSONALIZATION : 5201482.6122448975
TRAVEL_AND_LOCAL : 13984077.710144928
COMICS : 817657.2727272727
NEWS_AND_MAGAZINES : 9549178.467741935
TOOLS : 10801391.298666667
GAME : 15588015.603248259
BOOKS_AND_REFERENCE : 8767811.894736841
EDUCATION : 1833495.145631068
HOUSE_AND_HOME : 1331540.5616438356
ART_AND_DESIGN : 1986335.0877192982
SPORTS : 3638640.14285714

According to our analisys we got, that categories as:
* COMMUNICATION : 38 456 119 installs
* SOCIAL : 23 253 652 installs
* VIDEO_PLAYERS : 24 727 872 installs
* PHOTOGRAPHY : 17 840 110
* PRODUCTIVITY : 16 787 331
* GAME : 15 588 015
have most popular categories in Google Play market.
Let's analyse this categories detailed.

In [155]:
for app in free_eng_android:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                     or app[5] == '500,000,000+'
                                     or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])
    

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

Such apps as WhatsApp Messenger, Messenger – Text and Video Chat for Free,Skype - free IM & video calls have 1,000,000,000+ installs. 

In [156]:
for app in free_eng_android:
    if app[1] == 'SOCIAL' and (app[5] == '1,000,000,000+'
                                     or app[5] == '500,000,000+'
                                     or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Facebook : 1,000,000,000+
Facebook Lite : 500,000,000+
Tumblr : 100,000,000+
Pinterest : 100,000,000+
Google+ : 1,000,000,000+
Badoo - Free Chat & Dating App : 100,000,000+
Tango - Live Video Broadcast : 100,000,000+
Instagram : 1,000,000,000+
Snapchat : 500,000,000+
LinkedIn : 100,000,000+
Tik Tok - including musical.ly : 100,000,000+
BIGO LIVE - Live Stream : 100,000,000+
VK : 100,000,000+


Such apps as Facebook and Instagram have 1,000,000,000+ installs. 

In [157]:
for app in free_eng_android:
    if app[1] == 'VIDEO_PLAYERS' and (app[5] == '1,000,000,000+'
                                     or app[5] == '500,000,000+'
                                     or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

YouTube : 1,000,000,000+
Motorola Gallery : 100,000,000+
VLC for Android : 100,000,000+
Google Play Movies & TV : 1,000,000,000+
MX Player : 500,000,000+
Dubsmash : 100,000,000+
VivaVideo - Video Editor & Photo Movie : 100,000,000+
VideoShow-Video Editor, Video Maker, Beauty Camera : 100,000,000+
Motorola FM Radio : 100,000,000+


Such apps as YouTube and Google Play Movies & TV have 1,000,000,000+ installs. 

In [158]:
for app in free_eng_android:
    if app[1] == 'PHOTOGRAPHY' and (app[5] == '1,000,000,000+'
                                     or app[5] == '500,000,000+'
                                     or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

B612 - Beauty & Filter Camera : 100,000,000+
YouCam Makeup - Magic Selfie Makeovers : 100,000,000+
Sweet Selfie - selfie camera, beauty cam, photo edit : 100,000,000+
Google Photos : 1,000,000,000+
Retrica : 100,000,000+
Photo Editor Pro : 100,000,000+
BeautyPlus - Easy Photo Editor & Selfie Camera : 100,000,000+
PicsArt Photo Studio: Collage Maker & Pic Editor : 100,000,000+
Photo Collage Editor : 100,000,000+
Z Camera - Photo Editor, Beauty Selfie, Collage : 100,000,000+
PhotoGrid: Video & Pic Collage Maker, Photo Editor : 100,000,000+
Candy Camera - selfie, beauty camera, photo editor : 100,000,000+
YouCam Perfect - Selfie Photo Editor : 100,000,000+
Camera360: Selfie Photo Editor with Funny Sticker : 100,000,000+
S Photo Editor - Collage Maker , Photo Collage : 100,000,000+
AR effect : 100,000,000+
Cymera Camera- Photo Editor, Filter,Collage,Layout : 100,000,000+
LINE Camera - Photo editor : 100,000,000+
Photo Editor Collage Maker Pro : 100,000,000+


As we see here that the PHOTOGRAPHY niche has only one giant such as Google with 1,000,000,000+ installs, and all others have peer level of installs.

In [159]:
for app in free_eng_android:
    if app[1] == 'PRODUCTIVITY' and (app[5] == '1,000,000,000+'
                                     or app[5] == '500,000,000+'
                                     or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Microsoft Word : 500,000,000+
Microsoft Outlook : 100,000,000+
Microsoft OneDrive : 100,000,000+
Microsoft OneNote : 100,000,000+
Google Keep : 100,000,000+
ES File Explorer File Manager : 100,000,000+
Dropbox : 500,000,000+
Google Docs : 100,000,000+
Microsoft PowerPoint : 100,000,000+
Samsung Notes : 100,000,000+
SwiftKey Keyboard : 100,000,000+
Google Drive : 1,000,000,000+
Adobe Acrobat Reader : 100,000,000+
Google Sheets : 100,000,000+
Microsoft Excel : 100,000,000+
WPS Office - Word, Docs, PDF, Note, Slide & Sheet : 100,000,000+
Google Slides : 100,000,000+
ColorNote Notepad Notes : 100,000,000+
Evernote – Organizer, Planner for Notes & Memos : 100,000,000+
Google Calendar : 500,000,000+
Cloud Print : 500,000,000+
CamScanner - Phone PDF Creator : 100,000,000+


In [160]:
for app in free_eng_android:
    if app[1] == 'GAME' and (app[5] == '1,000,000,000+'
                                     or app[5] == '500,000,000+'
                                     or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Sonic Dash : 100,000,000+
PAC-MAN : 100,000,000+
Roll the Ball® - slide puzzle : 100,000,000+
Piano Tiles 2™ : 100,000,000+
Pokémon GO : 100,000,000+
Extreme Car Driving Simulator : 100,000,000+
Trivia Crack : 100,000,000+
Angry Birds 2 : 100,000,000+
Candy Crush Saga : 500,000,000+
8 Ball Pool : 100,000,000+
Subway Surfers : 1,000,000,000+
Candy Crush Soda Saga : 100,000,000+
Clash Royale : 100,000,000+
Clash of Clans : 100,000,000+
Plants vs. Zombies FREE : 100,000,000+
Pou : 500,000,000+
Flow Free : 100,000,000+
My Talking Angela : 100,000,000+
slither.io : 100,000,000+
Cooking Fever : 100,000,000+
Yes day : 100,000,000+
Score! Hero : 100,000,000+
Dream League Soccer 2018 : 100,000,000+
My Talking Tom : 500,000,000+
Sniper 3D Gun Shooter: Free Shooting Games - FPS : 100,000,000+
Zombie Tsunami : 100,000,000+
Helix Jump : 100,000,000+
Crossy Road : 100,000,000+
Temple Run 2 : 500,000,000+
Talking Tom Gold Run : 100,000,000+
Agar.io : 100,000,000+
Bus Rush: Subway Edition : 100,000,00

The Game niche has a high demand according to number of installs picture. But in same time has the high competition.

## Conclusion

In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both markets.

We concluded that photography niche has a good level of popularity and according to number of installs we understand that there is a lot of apps has the not high numbers of installs. It says that users do not prefare for specific application. 
It means we can deeply analyse the most popular photography apps in Google market, understand what users likes and dislikes on that apps, and create the new app with 'users likes'  
taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets. The markets are already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.